src/main/java/com/hivemq/cli/commands/options/TlsOptions.java

```diff
@@ -138,104 +127,255 @@ private boolean useBuiltSslConfig() {
+        final boolean certificateAndPrivateKeyExists = clientCertificatePath != null && clientPrivateKeyPath != null;
+        final boolean defaultKeystoreExists = defaultKeystore != null;
+        final boolean onlyDefaultCertificateExists =
+                defaultClientCertificateChain != null && defaultClientPrivateKey == null;
+        final boolean onlyDefaultPrivateKeyExists =
+                defaultClientCertificateChain == null && defaultClientPrivateKey != null;
+        final boolean defaultCertificateAndPrivateKeyExists =
+                defaultClientCertificateChain != null && defaultClientPrivateKey != null;
+
+        if (keystoreExists) {
+            if (certificateAndPrivateKeyExists) {
+                Log.warn("The keystore parameter shadows the private key corresponding certificate parameters. " +
+                        "If you want to use them instead please remove the keystore parameter.");
+            }
+            keyManagerFactory = TlsUtil.createKeyManagerFactoryFromKeystore(clientKeystorePath,
+                    clientKeystorePassword,
+                    clientKeystorePrivateKeyPassword);
+        } else if (onlyCertificateExists) {
+            throw new RuntimeException(
+                    "Only the client certificate parameter exists. Please add the private key parameter as well.");
+        } else if (onlyPrivateKeyExists) {
+            throw new RuntimeException(
+                    "Only the private key parameter exists. Please add the client certificate parameter as well.");
+        } else if (certificateAndPrivateKeyExists) {
+            final Collection<X509Certificate> x509Certificates =
+                    TlsUtil.getCertificateChainFromFile(clientCertificatePath);
+            final PrivateKey privateKey = TlsUtil.getPrivateKeyFromFile(clientPrivateKeyPath, clientPrivateKeyPassword);
+            final KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
+            ks.load(null, null);
+            final String createdKeystoreName = "privateKeyAlias";
+            final String createdKeystorePassword = "keystorePassword";
+            ks.setKeyEntry(createdKeystoreName,
+                    privateKey,
+                    createdKeystorePassword.toCharArray(),
+                    x509Certificates.toArray(new X509Certificate[0]));
+
+            keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            keyManagerFactory.init(ks, createdKeystorePassword.toCharArray());
+        } else if (defaultKeystoreExists) {
+            if (defaultCertificateAndPrivateKeyExists) {
+                Log.warn("The keystore default option (properties file) shadows the private key " +
+                        "corresponding certificate default options (properties file). " +
+                        "If you want to use them instead please remove the keystore default option.");
+            }
```

Comment:
I would throw an exception here. 
```suggestion
            if (defaultCertificateAndPrivateKeyExists) {
                throw new RuntimeException("Both 'auth.keystore' and 'auth.client.key' are configured in the defaults file '/path/to/default/file'.");
            }
```